src/game/engine/unit.cpp

```diff
@@ -436,7 +694,7 @@ BOOL UnitClass::Ok_To_Move(DirType dir)
     if (m_Rotating) {
         return false;
     }
-    if (dir - m_TurretFacing.Get_Current() != 0) {
+    if (m_TurretFacing.Difference_From_Current(dir)) {
```

Comment:
Should still check != 0 as its not a boolean value being checked. Be explicit so its clear at a glance and to less experienced programmers what you meant. Also, needs newline between scopes.